### PR TITLE
Move version number printing and include in detailed log

### PR DESCRIPTION
**Pull request by p-offtermatt**
~- [ ] Tests added for any new code~
- [x] Ran `make fmt-fix` (or had formatting run automatically on all files edited)
~- [ ] Documentation added for any new functionality~
- [x] Entry added to [UNRELEASED.md](./UNRELEASED.md) for any new functionality

### CODE REVIEW

**Review by konnov — APPROVED** [review]
Nice! There is probably a rare case, if a log misconfiguration happens and the users do not know the version number. Most likely this is not important though.

File: mod-tool/src/main/scala/at/forsyte/apalache/tla/Tool.scala [hunk 1]
```diff
@@ -443,11 +445,6 @@ object Tool extends LazyLogging {
   }
 
   private def printHeaderAndStatsConfig(): Unit = {
```
Comment (thpani) [inline mod-tool/src/main/scala/at/forsyte/apalache/tla/Tool.scala:447]:
```suggestion
  private def printStatsConfig(): Unit = {
```

**Review by thpani — APPROVED** [review]
LGTM! Thanks for implementing this!

File: mod-tool/src/main/scala/at/forsyte/apalache/tla/Tool.scala (hunk 1 above)
Comment (p-offtermatt) [reply mod-tool/src/main/scala/at/forsyte/apalache/tla/Tool.scala]: Thanks, good catch!

**Pull request merged**